pkg/pullsync/pullsync.go

```diff
@@ -112,6 +112,13 @@ func (s *Syncer) SyncInterval(ctx context.Context, peer swarm.Address, bin uint8
 	if err != nil {
 		return 0, 0, fmt.Errorf("new stream: %w", err)
 	}
+	defer func() {
+		if err != nil {
+			_ = stream.Reset()
+		} else {
+			_ = stream.FullClose()
```

Comment:
```suggestion
			go stream.FullClose()
```
It seems to me that we did not wait for conformation from the other side here, so i think we can do this in a separate go routine in order not to block the caller.